exla/lib/exla/defn.ex

```diff
@@ -1064,6 +1064,61 @@ defmodule EXLA.Defn do
     )
   end
 
+  defp to_operator(
+         :indexed_put,
+         [target, indices, updates],
+         %{type: type},
+         state
+       ) do
+    target = to_type(target, type)
+    updates = to_type(updates, type)
+
+    # Build update computation
+
+    subbuilder = subbuilder(state.builder, "scatter_reduction")
+
+    param_shape = EXLA.Shape.make_shape(type, {})
+    _left = EXLA.Op.parameter(subbuilder, 0, param_shape, "left")
+    right = EXLA.Op.parameter(subbuilder, 1, param_shape, "right")
+
+    scatter_fn = EXLA.Builder.build(right)
+
+    rank = target |> op_shape() |> tuple_size()
+    # indices_rank is guaranteed to be 2 by Nx.Shape
+    indices_rank = 2
+    rank_diff = rank - indices_rank + 1
+
+    indices_shape = op_shape(indices)
+
+    indices_shape =
+      [List.duplicate(1, rank_diff) | Tuple.to_list(indices_shape)]
+      |> List.flatten()
+      |> List.to_tuple()
+
+    indices = EXLA.Op.reshape(indices, indices_shape)
+
+    # If indices has shape {x, y}, updates is guaranteed by Nx.Shape to
+    # have shape {x}, so if we reshaped indices to {..., x, y}, we need to
+    # reshape updates to {..., x}
+
+    updates_shape = Tuple.delete_at(indices_shape, tuple_size(indices_shape) - 1)
+
+    updates = EXLA.Op.reshape(updates, updates_shape)
+
+    axes = axes_for_rank(rank)
+
+    EXLA.Op.scatter(
+      target,
+      indices,
+      updates,
+      scatter_fn,
+      rank,
+      [],
+      axes,
+      axes
+    )
```

Comment:
The only difference from `:indexed_add` is the `scatter_fn` right? We can move all of that to private `scatter` function and pass the built computation as an additional argument.

Reply:
Yes. That's what I meant by "DRYing".